src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-subsystems/nvmeof-subsystems.component.ts

```diff
@@ -135,17 +137,32 @@ export class NvmeofSubsystemsComponent extends ListWithDetails implements OnInit
     this.getSubsystems();
   }
 
-  getGatewayGroups() {
-    this.nvmeofService.listGatewayGroups().subscribe((response: CephServiceSpec[][]) => {
-      if (response?.[0].length) {
-        this.gwGroups = response[0].map((group: CephServiceSpec) => {
-          return {
-            content: group?.spec?.group
-          };
-        });
-      }
-      // Select first group if no group is selected
-      if (!this.group && this.gwGroups.length) this.onGroupSelection(this.gwGroups[0]);
-    });
-  }
+  // formats the gateway groups to be consumed for combobox item
+    formatGwGroupsList(data: CephServiceSpec[][]): ComboBoxItem[] {
+      return data[0].reduce((gwgrpList: ComboBoxItem[], group: CephServiceSpec) => {
+        if (group?.spec?.group) {
+          gwgrpList.push({
+            content: group.spec.group,
+          })
+        }
+        return gwgrpList;
+        },[])
+    }
+  
+    getGatewayGroups() {
+      this.nvmeofService.listGatewayGroups().subscribe((response: CephServiceSpec[][]) => {
+        if(response?.[0]?.length) {
+          this.gwGroups = this.formatGwGroupsList(response);
+        } else this.gwGroups = [];
+        // Select first group if no group is selected
+        if (!this.group && this.gwGroups.length) {
+          this.onGroupSelection(this.gwGroups[0]);
+          this.gwGroupsEmpty = false;
+        }
+        else {
+          this.gwGroupsEmpty = true;
+          this.gwGroupPlaceholder = $localize`No groups available`;
+        }
+      });
+    }
```

Comment:
Can we have a shared space for two same functions?

Reply:
Yes, this needs refactoring, did not want to make it in same PR as its a bug fix.
Its a debt from gw groups addition last release (which was unplanned before).

Reply:
@afreen23 I doubt we will circle back on this ever. That's why iftwould be better.

Reply:
:)  I moved `formatGwGroupsList` and for moving the `getGatewayGroups` we need a lot of refactoring so I will raise a PR for that before we merge this one.

Reply:
There is an update on this. We need to revamp the page and move out the gateway selector
updated this https://tracker.ceph.com/issues/71252